runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -884,11 +884,14 @@ TR_J9InlinerPolicy::genCodeForUnsafeGetPut(TR::Node* unsafeAddress,
    TR::TreeTop* nullComparisonTree = nullComparisonBlock->getLastRealTreeTop();
    TR::TreeTop *nullComparisonEntryTree = nullComparisonBlock->getEntry();
    TR::TreeTop *nullComparisonExitTree = nullComparisonBlock->getExit();
-   //if conversionNeeded is true, we haven't generated and we don't need arrayDirectAccessBlock
-   TR::Block *arrayDirectAccessBlock = conversionNeeded ? nullComparisonTree->getNode()->getBranchDestination()->getNode()->getBlock() : NULL;
+
+   bool arrayBlockNeeded = conversionNeeded || (comp()->fej9()->isOffHeapAllocationEnabled() && comp()->target().is64Bit());
```

Comment:
This was (or will be as part of newer vm/gc chagnes) moved to object model. You can use `TR::Compiler->om.isOffHeapAllocationEnabled()` instead but you don't need to guard with any macro as it will just return false in all other cases.